app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/AuthActionCustomHeader.scala

```diff
@@ -75,14 +75,14 @@ class AuthActionSubmitterHeader @Inject()(customsAuthService: CustomsAuthService
 
   override def eitherCspAuthData[A](maybeNrsRetrievalData: Option[NrsRetrievalData])(implicit vhr: HasRequest[A] with HasConversationId): Either[ErrorResponse, AuthorisedAsCsp] = {
     val cpsAuth: Either[ErrorResponse, Csp] = for {
-      maybeBadgeId <- eitherBadgeIdentifier(allowNone = true).right
-      maybeEori <- eitherEori.right
+      maybeBadgeId <- eitherBadgeIdentifier(allowNone = true)
+      maybeEori <- eitherEori
     } yield {
       logger.info(headerValidator.logEoriAndBadgeIdHeaderText(XSubmitterIdentifierHeaderName, maybeBadgeId, maybeEori))
       Csp(maybeEori, maybeBadgeId, maybeNrsRetrievalData)
     }
 
-    if (cpsAuth.isRight && cpsAuth.right.get.badgeIdentifier.isEmpty && cpsAuth.right.get.eori.isEmpty) {
+    if (cpsAuth.isRight && cpsAuth.toOption.get.badgeIdentifier.isEmpty && cpsAuth.toOption.get.eori.isEmpty) {
```

Comment:
We should refactor this to more idiomatic Scala. We shouldn't have to call `get` if we can help it.